discovery-provider/src/utils/get_all_other_nodes.py

```diff
@@ -153,9 +133,23 @@ def get_all_other_content_nodes_cached(redis) -> List[Dict[str, str]]:
     return get_json_cached_key(redis, ALL_CONTENT_NODES_CACHE_KEY)
 
 
-def get_all_alive_content_nodes_cached(redis) -> List[Dict[str, str]]:
+def get_all_healthy_content_nodes_cached(redis) -> List[Dict[str, str]]:
     """
     Attempts to get the healthy content nodes from redis.
     """
 
-    return get_json_cached_key(redis, ALL_ALIVE_CONTENT_NODES_CACHE_KEY)
+    return get_json_cached_key(redis, ALL_HEALTHY_CONTENT_NODES_CACHE_KEY)
+
+
+async def get_node_if_healthy(content_node: Dict[str, str]):
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(
+                f"{content_node['endpoint']}/health_check"
+            ) as response:
+                return content_node if response.status == 200 else None
```

Comment:
other checks would go here if it needs to be stricter - like disk space, min version, etc. if we do it that way, I think it would be best to pass as queryparams like `maxDiskUsage=97` and CN can just return non-200